### PR TITLE
DS_Week2_복습과제_여채윤

### PR DESCRIPTION
**Pull request by chaeyunyeo**

### CODE REVIEW

**Comment by kkyung0131** [comment]
1.3
열의 이름을 알고 싶을 때 
데이터명.columns로 확인할 수 있습니다!(ex. train_df.columns)

1.5
train_df[‘quality’].value_counts( )를 통해서도 원하는 결과를 얻을 수 있습니다!

1.6 
더 간단하게 인코딩하신 것 같아요!
레이블 인코딩을 할 수 있는 다른 코드도 보시면 좋을 것 같아 적어드립니다!
1)(파머완 코드)
items = ['white','red']
encoder = LabelEncoder()
encoder.fit(items)
labels = encoder.transform(items)
train_df['type'] = encoder.fit_transform(train_df['type'].values)
print(train_df['type'].value_counts())

2)(차수빈벗 코드)
encoder = LabelEncoder() # 라벨 인코더 객체 생성
encoder.fit(train_df['type'])
train_df['type'] = encoder.transform(train_df['type']) # label-encoding을 시킨 결과를 다시 df의 컬럼에 저장
train_df['type'].value_counts()
# red, white -> 0,1로 encoding

3)(조혜빈벗 코드)
items=['type']
le=LabelEncoder()
for i in items:
traindrop[i]=le.fit_transform(traindrop[i])
traindrop['type'].value_counts()  

1.8
정확도를 소수점 4자리까지 뽑고 싶다면
print('결정 트리 예측 정확도: {0:.4f}'.format(accuracy))
으로 작성할 수 있습니다!
2.1, 2.2
답하고 코드가 똑같으신데, 답이 조금 다른 게 이상하네요ㅠㅠ 저는 아무리 살펴봐도 차이를 못 찾겠어서, 혹시 발견했다면 알려주시면 감사하겠습니다!!
2.5
저랑 정확도가 다르게 나와서 살펴봤더니, random_state를 2.4번과는 다르게 지정 안해주셔서 그런 것 같아요! 지금 있는 코드에서 random_state=12만 추가해주시면 될 것 같습니다!!
rf_clf1 = RandomForestClassifier(n_estimators=50, min_samples_leaf=8,max_depth=14,
min_samples_split=8, random_state=12)
 
2.6
wine 데이터의 경우 파마완에서 사용한 데이터와는 다르게 변수의 개수가 몇 개 안되기 때문에 막대 그래프 상위 20개만 보여달라고 할 필요는 없을 것 같아서 해당 필요 없는 부분만 삭제해주시면 좋을 것 같아요.
ftr_top = ftr_importances.sort_values(ascending=False)
ftr_top = ftr_importances.sort_values(ascending=False)
3.2
선형 SVM 모델이므로 kernel 파라미터를 ‘linear’로 설정해주셔야 해요! 문제에 언급된 부분이 없음에도 cv를 설정해주신 점은 좋습니다!

3.3
best_estimator_를 이용하는 방법 말고 다른 방법도 있어서 적어드립니다.
lin_svm = SVC(kernel='linear',C=1)
lin_svm.fit(X_train,y_train)
pred = lin_svm.predict(X_test)
accuracy = accuracy_score(y_test, pred)
print('accuracy : {0:.4f}'.format(accuracy))
 
3.4
GridSearchCV에서 param 인자가 잘못 설정된 것 같아요. param 대신 params를 넣어주셔야 합니다. 변수명 확인 부탁드려요!
